libs/store_helper/ebpf_store_helper.c

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#ifndef USER_MODE
+#include "kernel\ebpf_registry_helper_km.h"
+#else
+#include "user\ebpf_registry_helper_um.h"
+#endif
+
+#include "ebpf_program_types.h"
+#include "ebpf_store_helper.h"
+#include "ebpf_windows.h"
+
+uint32_t
+ebpf_store_open_or_create_provider_registry_key(_Out_ ebpf_registry_key_t* provider_key)
+{
+    __return_type status = _SUCCESS;
+    ebpf_registry_key_t root_key = NULL;
+    *provider_key = NULL;
+
+    // Open (or create) root eBPF registry path.
+#ifdef USER_MODE
```

Comment:
We try to avoid ifdefs like this whenever possible, and here I think it's possible.

Reply:
This was the existing implementation, which distinguishes between `EBPF_ROOT_RELATIVE_PATH` & `EBPF_ROOT_REGISTRY_PATH`.

Reply:
I get the "this was the existing implementation" answer and so this is not a blocking comment.
But I don't understand (and the comments in the code don't explain) why we need to distinguish between EBPF_ROOT_RELATIVE_PATH & EBPF_ROOT_REGISTRY_PATH in this function.

Reply:
@saxena-anurag can you recall the history of this spec?

Reply:
Still haven't seen an answer to this.